### ISSUE TITLE
ChatScript crashes on specific words

### ISSUE THREAD

**Issue by jb-kerboeuf** [bwilcox-1234/ChatScript#64]
Hi !
The program crashes when certain words are in the input.

With language=FRENCH if I say "chocolate" or "building" at any point in the conversation, the program crashes.

I tried :prepare chocolate to see what’s wrong and it gave me this and then it crashed again.
```
TokenControl: DO_SUBSTITUTE_SYSTEM DO_NUMBER_MERGE DO_SPELLCHECK


Original User Input: chocolate
Tokenized into: chocolate
Actual used input: chocolate


Concepts:

1: chocolate (raw):  //
1: chocolate (canonical):  //

Sequences:
 +chocolate +~flavors  
```

**Comment by kuzyn** [comment]
Hi @jb-kerboeuf maybe you can [trace](https://github.com/bwilcox-1234/ChatScript/blob/master/WIKI/ChatScript-Debugging-Manual.md#trace) the crash and post the  offending lines here?

Also, do you get any errors on `:build` ?

**Comment by ludan-av** [comment]
complement:
- With 7.2 out of the box version: I double checked and have the problem. It crashes.
- When I comment all the stuff in files0.txt and `:build 0`: it doesn't crash no more (and spellcheck changes "chocolate" to "chocolaté", which looks fine).

**Comment by kuzyn** [comment]
Yep, can replicate. From the user log:

```
Command: :prepare chocolate
TokenControl: DO_SUBSTITUTE_SYSTEM DO_NUMBER_MERGE DO_PROPERNAME_MERGE DO_DATE_MERGE DO_SPELLCHECK DO_INTERJECTION_SPLITTING DO_PARSE 


Original User Input: chocolate
Tokenized into: chocolate  
Actual used input: chocolate 

  *** Sentence start:  (0) to  (0)

------------- POS rules pass 1: 

POS 1 words: chocolate ()  

---- Assign roles
     Zones: Ambiguous zone, 
  *** Sentence start: chocolate (1) to chocolate (1)
1) "raw: chocolate canonical: Missing_canonical" ()
    need 1  MainSubject MainVerb Main: - s:0 aux:0 v:0 io:0 o:0
      - leftover want 1:  MainSubject MainVerb Main:
        ->  subject:  verb:  indirectobject:  object:  lastclause@:  lastverbal@:
PreFinishSentence: chocolate () 
Zones: Ambiguous zone, 
badparse: no main verb
Not a sentence
Xref: 1:chocolate   
1:chocolate   
badparse Tagged POS 1 words: chocolate ()  
  MainSentence:  PRESENT 


Concepts: 

1: chocolate (raw):  +~noun +~sentenceend +chocolate +~flavors
```

**Comment by bwilcox-1234** [comment]
fixed in src, not recompiled, not in 7.31
